Analysis/PhotonEnergyCorrect/include/photonCorrector.h

```diff
@@ -0,0 +1,207 @@
+#ifndef photonCorrector_h
+#define photonCorrector_h 1
+
+#include "EVENT/ReconstructedParticle.h"
+#include <cassert>
+
+class photonCorrector {
+ public:
+  photonCorrector() {_initialized=false;}
+  ~photonCorrector() {}
+
+  enum { set_std, set_noInterMod };
+
+  void setDefaultValues(int defSet) {
+    if ( defSet == set_std ) {
+      setDefaultValues_interModBruteForceCorr();
+    } else if ( defSet == set_noInterMod ) {
+      setDefaultValues_no_interModBruteForceCorr();
+    } else {
+      std::cout << "photonCorrector: unknown default param set! giving up!" << std::endl;
+      assert(0);
+    }
+    return;
+  }
+
+  void setDefaultValues_interModBruteForceCorr() {
+    // this one is for the case in which inter-module gaps are corrected by "brute force" at hit reco level
+    // determined for ILD_l5_o1_v02 model
+
+    _barrel_limit=0.75;
+    _endcap_limit=0.8;
+
+    _energyLin_const = 9.87661e-01;
+    _energyLin_logen = 1.40676e-02;
+
+    _phiBarrelCorr_pos_const = 0.411540;
+    _phiBarrelCorr_pos_logen = 0.0149989;
+    _phiBarrelCorr_depth     = -0.0923036;
+    _phiBarrelCorr_width1    =  0.014018;
+    _phiBarrelCorr_width2    = 0.039961;
+
+    _costhCorr_gaus1_norm_const = -7.97130e-02;
+    _costhCorr_gaus1_norm_logen = 5.85799e-02;
+    _costhCorr_gaus1_mean = 0.235;
+    _costhCorr_gaus1_sigm = 0.01;
+
+    _costhCorr_gaus2_norm_const = -3.52624e-02;
+    _costhCorr_gaus2_norm_logen = 3.07763e-02;
+    _costhCorr_gaus2_mean = 0.588;
+    _costhCorr_gaus2_sigm = 0.009;
+
+    _costhCorr_gaus3_norm = -0.0422968;
+    _costhCorr_gaus3_mean = 0.774;
+    _costhCorr_gaus3_sigm = 0.009;
+
+    _costhCorr_endcap_scale = 1.002;
+
+    _endcap_gaus1_norm = -0.025;
+    _endcap_gaus1_mean = 855.  ;
+    _endcap_gaus1_sigm = 23.   ;
+
+    _endcap_gaus2_norm = -0.07 ;
+    _endcap_gaus2_mean = 1489. ;
+    _endcap_gaus2_sigm = 18.   ;
+
+    // actual values not important, but should use same defs when determining the correction and applying it
+    _assumed_boxsize=400; // size of inner endcap box (ECAL plug/ring)
```

Comment:
These two parameters can be taken from the geometry itself right ?

Reply:
now take geom info from dd4hep as much as possible